direct/io-kafka/src/main/java/cz/o2/proxima/direct/kafka/Consumers.java

```diff
@@ -119,7 +120,7 @@ public boolean consumeWithConfirm(
       }
       committed.compute(tp.partition(), (k, v) -> v == null || v <= offset ? offset + 1 : v);
       committer.confirm(tp, offset);
-      return true;
+      return watermark < Watermarks.MAX_WATERMARK;
```

Comment:
We must move this out of this PR or add a test for that.

Reply:
Also cover in Kafka IT